src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```diff
@@ -374,6 +375,16 @@ private static BaseXEStore CreateXEventStore(ConnectionInfo connInfo, SqlStoreCo
             return store;
         }
 
+        /// <summary>
+        /// Nulls out properties in ConnectionDetails that aren't compatible with XElite.
+        /// </summary>
+        private static ConnectionDetails CreateXEliteConnectionDetails(ConnectionDetails connDetails){
+            connDetails.ConnectRetryCount = null;
+            connDetails.ConnectRetryInterval = null;
+            connDetails.MultiSubnetFailover = null;
+            return connDetails;
```

Comment:
Given that we're modifying the existing object this doesn't really need to return it. Maybe just call this `RemoveIncompatibleConnectionProperties` and have the return type be void.